source/by-role/developer/common/adding-a-new-language.rst

```diff
@@ -0,0 +1,217 @@
+      <script src="/studio/static-assets/components/cstudio-common/resources/kr/base.js"></script>
+      <script src="/studio/static-assets/components/cstudio-common/resources/es/base.js"></script>
+      <script src="/studio/static-assets/components/cstudio-common/resources/de/base.js"></script>
+      <script src="/studio/static-assets/components/cstudio-common/resources/ja/base.js"></script>
+
+.. code-block:: html
+      :force:
+      :linenos:
+      :emphasize-lines: 5
+      :caption: *studio-ui/templates/web/legacy-site-config.ftl*
+
+      <script type="text/javascript" src="/studio/static-assets/components/cstudio-common/resources/en/base.js"></script>
+      <script type="text/javascript" src="/studio/static-assets/components/cstudio-common/resources/kr/base.js"></script>
+      <script type="text/javascript" src="/studio/static-assets/components/cstudio-common/resources/es/base.js"></script>
+      <script type="text/javascript" src="/studio/static-assets/components/cstudio-common/resources/de/base.js"></script>
+      <script type="text/javascript" src="/studio/static-assets/components/cstudio-common/resources/ja/base.js"></script>
+
+
+---------------------------------------------------------------
+5. Add the Translations File for Legacy Forms and Content Types
+---------------------------------------------------------------
+* To add the translations file for legacy forms and content types, in your ``studio-ui`` code, navigate to ``studio-ui/static-assets/components/cstudio-common/resources/``.  Create a folder using the two letter language code for the new language being added, ``studio-ui/static-assets/components/cstudio-common/resources/ja``
+* Copy the file ``studio-ui/static-assets/components/cstudio-common/resources/en/base.js`` and paste it into the newly created folder
+* Start translating the content in ``studio-ui/static-assets/components/cstudio-common/resources/ja/base.js`` and save your changes
+
+    .. code-block:: js
+       :caption: *studio-ui/static-assets/components/cstudio-common/resources/ja/base.js*
+       :linenos:
+
+       CStudioAuthoring.Messages.registerBundle("siteDashboard", "ja", {
+       dashboardTitle: "ダッシュボード",
+
+       dashboardCollapseAll: "すべて折りたたむ",
+       ...
+
+Remember to change the language code in the all the ``registerBundle`` calls in the ``base.js`` file
+
+  .. code-block:: js
+
+     CStudioAuthoring.Messages.registerBundle("dialogs", "ja", {
+
+--------------------------------------
+6. Build, deploy and test your changes
+--------------------------------------
+Don't forget to build and deploy.  Before building, remember to run prettier on the file ``ui/app/src/utils/i18n.ts``
+
+.. code-block:: bash
+    :caption: *src/studio-ui*
+
+    prettier --write ui/app/src/utils/i18n.ts
+
+Or, if you have ``prettier`` installed via npm, run the following:
+
+.. code-block:: bash
+    :caption: *Run prettier using npm*
+
+    npx prettier ui/app/src/utils/i18n.ts --write
```

Comment:
Prettier is on studio ui's env, and it should be used there through so it picks our config. Or better yet, for certainty, specify the config:
```sh
cd ui/app
yarn prettier --config ../../prettier.config.js --write ./src/utils/i18n.ts
```